### PR TITLE
Unify jvm memory settings validation & improve upload error message

### PR DESCRIPTION
**Pull request by juhoautio**
Users shouldn't need to go view Azkaban source to find out why project upload failed in this kind of a case (from azkaban mailing list):

> I am trying to upload a zip project file and getting this error, could not figure out the reason.

> Users were getting java heap space error very often and by mistake while trying to solve it I have introduced the "job.max.Xmx=" twice in azkaban.properties, one of them empty.

```
Installation Failed.
For input string: "" 

2019/01/24 13:25:02.852 -0300 INFO [ProjectManagerServlet] [Azkaban] Upload: reference of project test_project_for_delete is 101056896
2019/01/24 13:25:02.853 -0300 INFO [ProjectManagerServlet] [Azkaban] Uploading file lte.zip
2019/01/24 13:25:02.853 -0300 INFO [AzkabanProjectLoader] [Azkaban] Uploading files to test_project_for_delete
2019/01/24 13:25:02.864 -0300 WARN [XmlValidatorManager] [Azkaban] Validator directory validators does not exist or is not a directory.
2019/01/24 13:25:02.865 -0300 WARN [XmlValidatorManager] [Azkaban] Azkaban properties file does not contain the key project.validators.xml.file
2019/01/24 13:25:02.865 -0300 INFO [AzkabanProjectLoader] [Azkaban] Validating project lte.zip using the registered validators []
2019/01/24 13:25:02.875 -0300 INFO [ProjectManagerServlet] [Azkaban] Installation Failed.
java.lang.NumberFormatException: For input string: ""
        at java.base/java.lang.NumberFormatException.forInputString(NumberFormatException.java:65)
        at java.base/java.lang.Long.parseLong(Long.java:702)
        at java.base/java.lang.Long.parseLong(Long.java:817)
        at azkaban.utils.Utils.parseMemString(Utils.java:453)
        at azkaban.project.FlowLoaderUtils.checkJobProperties(FlowLoaderUtils.java:260)
        at azkaban.project.DirectoryFlowLoader.loadProjectFlow(DirectoryFlowLoader.java:142)
        at azkaban.project.AzkabanProjectLoader.uploadProject(AzkabanProjectLoader.java:105)
        at azkaban.project.ProjectManager.uploadProject(ProjectManager.java:506)
        at azkaban.webapp.servlet.ProjectManagerServlet.ajaxHandleUpload(ProjectManagerServlet.java:1753)
        at azkaban.webapp.servlet.ProjectManagerServlet.handleUpload(ProjectManagerServlet.java:1836)
        at azkaban.webapp.servlet.ProjectManagerServlet.handleMultiformPost(ProjectManagerServlet.java:201)
        at azkaban.webapp.servlet.LoginAbstractAzkabanServlet.doPost(LoginAbstractAzkabanServlet.java:311)
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:688)
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:770)
        at org.mortbay.jetty.servlet.ServletHolder.handle(ServletHolder.java:511)
        at org.mortbay.jetty.servlet.ServletHandler.handle(ServletHandler.java:401)
        at org.mortbay.jetty.servlet.SessionHandler.handle(SessionHandler.java:182)
        at org.mortbay.jetty.handler.ContextHandler.handle(ContextHandler.java:766)
        at org.mortbay.jetty.handler.HandlerWrapper.handle(HandlerWrapper.java:152)
        at org.mortbay.jetty.Server.handle(Server.java:326)
        at org.mortbay.jetty.HttpConnection.handleRequest(HttpConnection.java:542)
        at org.mortbay.jetty.HttpConnection$RequestHandler.content(HttpConnection.java:945)
        at org.mortbay.jetty.HttpParser.parseNext(HttpParser.java:756)
        at org.mortbay.jetty.HttpParser.parseAvailable(HttpParser.java:218)
        at org.mortbay.jetty.HttpConnection.handle(HttpConnection.java:404)
        at org.mortbay.jetty.bio.SocketConnector$Connection.run(SocketConnector.java:228)
        at org.mortbay.thread.QueuedThreadPool$PoolThread.run(QueuedThreadPool.java:582)
2019/01/24 13:25:02.877 -0300 INFO [ProjectManagerServlet] [Azkaban] Upload: project test_project_for_delete version is 0, reference is 101056896
2019/01/24 13:25:03.281 -0300 INFO [LoginAbstractAzkabanServlet] [Azkaban] 10.245.221.148 username "GET /manager project=test_project_for_delete HTTP/1.1" browser
```

### CODE REVIEW

**Comment by juhoautio** [comment]
@burgerkingeater are you able to review this, too?

**Comment by burgerkingeater** [comment]
@juhoautio can you help me better understand this PR? Why would it help the described issue?

**Comment by juhoautio** [comment]
If there's an empty value like `"job.max.Xmx="` in some properties, upload fails with this error:

> Installation Failed.
> For input string: ""

So it doesn't help much because it doesn't even tell the name of the problematic property.

This PR improves it by returning a better error message ie. one that includes the property name.

Currently user would need to grab the stack trace, get azkaban source code, and find the line, to know which property caused the upload failure.

File: az-core/src/main/java/azkaban/utils/MemConfValue.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
+package azkaban.utils;
+
+import azkaban.Constants;
+import com.google.common.base.Preconditions;
+import org.apache.commons.lang.StringUtils;
+
+public class MemConfValue {
```
Comment (burgerkingeater) [inline az-core/src/main/java/azkaban/utils/MemConfValue.java]: Can this be applied to other properties? If so, can we make this universal?

Comment (juhoautio) [reply az-core/src/main/java/azkaban/utils/MemConfValue.java]: How? I guess these are the only properties that azkaban parses expecting this pattern.

**Comment by juhoautio** [comment]
@burgerkingeater ..good to merge?

**Pull request merged**